include/cuco/static_multimap.cuh

```diff
@@ -0,0 +1,1808 @@
+   */
+  template <typename InputIt, typename OutputIt, typename KeyEqual = thrust::equal_to<key_type>>
+  OutputIt retrieve(InputIt first,
+                    InputIt last,
+                    OutputIt output_begin,
+                    cudaStream_t stream = 0,
+                    KeyEqual key_equal  = KeyEqual{}) const;
+
+  /**
+   * @brief Retrieves all the matches corresponding to all keys in the range `[first, last)`.
+   *
+   * The `_outer` suffix signifies that non-matches are included in the output: If the key
+   * `k = *(first + i)` exists in the map, copies `k` and all associated values to unspecified
+   * locations in `[output_begin, output_end)`. Else, copies `k` and the empty value sentinel.
+   *
+   * Behavior is undefined if the total number of matching keys exceeds `std::distance(output_begin,
+   * output_end)`. Use `count_outer()` to determine the number of matching keys.
+   *
+   * @tparam InputIt Device accessible input iterator whose `value_type` is
+   * convertible to the map's `key_type`
+   * @tparam OutputIt Device accessible output iterator whose `value_type` is
+   * convertible to the map's `value_type`
+   * @tparam KeyEqual Binary callable type
+   * @param first Beginning of the sequence of keys
+   * @param last End of the sequence of keys
+   * @param output_begin Beginning of the sequence of key/value pairs retrieved for each key
+   * @param stream CUDA stream used for retrieve_outer
+   * @param key_equal The binary function to compare two keys for equality
+   * @return The iterator indicating the last valid key/value pairs in the output
+   */
+  template <typename InputIt, typename OutputIt, typename KeyEqual = thrust::equal_to<key_type>>
+  OutputIt retrieve_outer(InputIt first,
+                          InputIt last,
+                          OutputIt output_begin,
+                          cudaStream_t stream = 0,
+                          KeyEqual key_equal  = KeyEqual{}) const;
+
+  /**
+   * @brief Retrieves all pairs matching the input probe pair in the range `[first, last)`.
+   *
+   * The `pair_` prefix indicates that the input data type is convertible to the map's
+   * `value_type`. If pair_equal(*(first + i), slot[j]) returns true, then *(first+i) is
+   * stored to `probe_output_begin`, and slot[j] is stored to `contained_output_begin`.
+   *
+   * Behavior is undefined if the total number of matching pairs exceeds
+   * `std::distance(probe_output_begin, probe_output_end)` (or
+   * `std::distance(contained_output_begin, contained_output_end)`). Use
+   * `pair_count()` to determine the number of matching pairs.
+   *
+   * @tparam InputIt Device accessible random access input iterator where
+   * `std::is_converitble<std::iterator_traits<InputIt>::value_type,
+   * static_multimap<K, V>::value_type>` is `true`
+   * @tparam OutputZipIt1 Device accessible output zip iterator for probe matches
+   * @tparam OutputZipIt2 Device accessible output zip iterator for contained matches
```

Comment:
These don't need to be zip iterators, their value type just needs to be constructible from the maps value type. 
```suggestion
   * @tparam OutputIt1 Device accessible output iterator whose `value_type` is constructible from `InputIt`s `value_type`.
   * @tparam OutputIt2 Device accessible output iterator whose `value_type` is constructible from the map's `value_type`. 
```

Reply:
Technically, the `value_type` of `InputIt` and the map's `value_type` can be different. Can you double check that the implementation supports that?